packages/engine/src/output/buffer/util.rs

```diff
@@ -1,42 +1,74 @@
-use std::path::PathBuf;
-
 use memory::shared_memory::MemoryId;
 
-use crate::{
-    output::{buffer::RELATIVE_PARTS_FOLDER, error::Result},
-    proto::ExperimentId,
-};
+use crate::{output::buffer::RELATIVE_PARTS_FOLDER, proto::ExperimentId};
+
+// TODO: move this to top-level
+#[derive(PartialEq, Eq, Clone)]
+pub enum EngineExitStatus {
```

Comment:
Why do we have this enum? I find it very confusing when we already have too many `Status` structs such as `SimStatus` and `EngineStatus`.

Reply:
I also don't see what information this enum actually encapsulates

Reply:
It looks like it's only used in the cleanup function for warning messages. I don't see this as particularly useful.

Reply:
We used to have a `bool` instead of this. It should be more than 2 states at some point. (like `Panic`)

Reply:
I still don't see the use of it, it seems to only be used for warning messages when removing files, and I don't get how the exit 'status' affects those warning messages.

Reply:
To me if the files are present after a successful run it's a bigger issue than after a crash.
Currently destructors don't properly cleanup in case something bad happens. So having files left is expected and this PR adds a cleanup for them.
But since `cleanup_experiment` is also used in the happy path, if there are files left at the end of a successful run it's pretty suspicious.